src/tools/mesh_tester.cpp

```diff
@@ -68,53 +60,108 @@ struct Input
-    skipWeld(false),
-    errorCode(SUCCESS)
-  { };
 
-  Input(int argc, char** argv);
+  Input()
+    : stlInput("")
+    , vtkOutput("")
+    , resolution(0)
+    , weldThreshold(1e-6)
+    , skipWeld(false)
+  { };
 
-  void showhelp()
-  {
-    std::cout
-      << "Argument usage:"
-       "\n  --help           Show this help message."
-       "\n  --resolution N   Resolution of uniform grid.  Default N = 0."
-       "\n                   - Set to 1 to run the naive algorithm, without"
-       "\n                   a spatial index."
-       "\n                   - Set to less than 1 to use the spatial index"
-       "\n                   with a resolution of the cube root of the"
-       "\n                   number of triangles."
-       "\n  --infile fname   The STL input file (must be specified)."
-       "\n  --outfile fname  Output file name for collisions and welded mesh"
-       "\n                   (defaults to a file in the CWD with the input file name)"
-       "\n                   Collisions mesh will end with '.collisions.vtk' and"
-       "\n                   welded mesh will end with '.welded.vtk'."
-       "\n  --weldThresh eps Distance threshold for welding vertices. "
-       "\n                   Default: eps = 1e-6"
-       "\n  --skipWeld       Don't weld vertices (useful for testing,"
-       "\n                   not helpful otherwise)."
-      << std::endl << std::endl;
-  };
+  void parse(int argc, char** argv, CLI::App& app);
 
   std::string collisionsMeshName() { return vtkOutput + ".collisions.vtk"; }
   std::string collisionsTextName() { return vtkOutput + ".collisions.txt"; }
   std::string weldMeshName() { return vtkOutput + ".weld.vtk"; }
+
+private:
+  void fixOutfilePath();
 };
 
+void Input::parse(int argc, char** argv, CLI::App& app)
+{
+  app.add_option("-r,--resolution", resolution,
+                 "Resolution of uniform grid. \n"
+                 "Set to 1 to run the naive algorithm without a spatial index. \n"
+                 "Set to less than 1 to use the spatial index with a resolution \n"
+                 "of the cube root of the number of triangles.")
+  ->capture_default_str();
+
+  app.add_option("-i,--infile", stlInput,"The STL input file")
+  ->required()
+  ->check(CLI::ExistingFile);
+
+  app.add_option("-o,--outfile", vtkOutput,
+                 "Output file name for collisions and welded mesh. \n"
+                 "Defaults to a file in the CWD with the input file name. \n"
+                 "Collisions mesh will end with '.collisions.vtk' and \n"
+                 "welded mesh will end with '.welded.vtk'.");
+
+  app.add_option("--weldThresh", weldThreshold,
+                 "Distance threshold for welding vertices.")
+  ->capture_default_str();;
+
+  app.add_flag("--skipWeld", skipWeld,
+               "Don't weld vertices (useful for testing, not helpful otherwise).");
+
+  app.get_formatter()->column_width(35);
+
+  // Could throw an exception
+  app.parse(argc, argv);
```

Comment:
Here is the generated usage output:
```
>./bin/mesh_tester  --help
MeshTester example
Usage: ./bin/mesh_tester [OPTIONS]

Options:
  -h,--help                        Print this help message and exit
  -r,--resolution INT=0            Resolution of uniform grid.
                                   Set to 1 to run the naive algorithm without a spatial index.
                                   Set to less than 1 to use the spatial index with a resolution
                                   of the cube root of the number of triangles.
  -i,--infile TEXT:FILE REQUIRED   The STL input file
  -o,--outfile TEXT                Output file name for collisions and welded mesh.
                                   Defaults to a file in the CWD with the input file name.
                                   Collisions mesh will end with '.collisions.vtk' and
                                   welded mesh will end with '.welded.vtk'.
  --weldThresh FLOAT=1e-06         Distance threshold for welding vertices.
  --skipWeld                       Don't weld vertices (useful for testing, not helpful otherwise).
```

Reply:
And this is the output if you supply an invalid file:
```
>./bin/mesh_tester  --infile foo
--infile: File does not exist: foo
Run with --help for more information.
```